packages/babel-preset-react/src/index.js

```diff
@@ -25,35 +22,28 @@ export default declare((api, opts) => {
     pragmaFrag = pragmaFrag || "React.Fragment";
   }
 
-  // TODO: (Babel 8) Don't cast these options but validate it
+  // TODO: (Babel 8) Don't cast this option but validate it
   const development = !!opts.development;
-  const useBuiltIns = !!opts.useBuiltIns;
-
-  const transformReactJSXPlugin =
-    runtime === "automatic" && development
-      ? transformReactJSXDevelopment
-      : transformReactJSX;
 
   return {
     plugins: [
       [
-        transformReactJSXPlugin,
+        development ? transformReactJSXDevelopment : transformReactJSX,
         {
           importSource,
           pragma,
           pragmaFrag,
           runtime,
           throwIfNamespace,
-          useBuiltIns,
-          useSpread,
           pure,
+          // TODO (Babel 8): Remove `useBuiltIns` & `useSpread`
+          useBuiltIns: !!opts.useBuiltIns,
+          useSpread:
+            "useSpread" in opts ? opts.useSpread : runtime !== "classic",
```

Comment:
This can simply be `opts.useSpread`, unless I'm missing something: it defaults to `undefined`, which is falsy with the `classic` runtime and ignored with the `automatic` one.